userspace/libpman/src/lifecycle.c

```diff
@@ -48,18 +47,6 @@ static void pman_save_attached_progs()
 	g_state.attached_progs_fds[7] = bpf_program__fd(g_state.skel->progs.pf_kernel);
 #endif
 	g_state.attached_progs_fds[8] = bpf_program__fd(g_state.skel->progs.signal_deliver);
-
-	for(int j = 0; j < MODERN_BPF_PROG_ATTACHED_MAX; j++)
```

Comment:
@Andreagit97 mind getting me up to speed wrt the reason for changing the logic above to

```
for(int j = 0; j < MODERN_BPF_PROG_ATTACHED_MAX; j++)
{
      g_state.attached_progs_fds[j] = -1;
}
```

and below we have 

```
for(int j = 0; j < MODERN_BPF_PROG_ATTACHED_MAX; j++)
{
	if(g_state.attached_progs_fds[j] != -1)
	{
		nprogs_attached++;
	}
}
```

Besides this question, LGTM!

Reply:
Ei! The idea here was to move all the logic inside `pman_get_metrics_v2` in this way we could avoid creating a global variable `g_state.n_attached_progs` and just use a local variable. In the end, the only place where we need this information regarding attached_progs is inside `pman_get_metrics_v2`. Now the modern ebpf and the legacy one do the same loop in the same place so it should be easily to maintain in the future